src/junit.rs

```diff
@@ -1,4 +1,4 @@
-use pyo3::exceptions::PyException;
+use pyo3::exceptions::{PyException, PyRuntimeError};
```

Comment:
I'll leave some comments for me and other questions as I'm learning Rust on the side so not an absolute nooooob but using this as an opportunity for us to learn more by chatting 😬 lmk if it's too exhaustive.

So the main use of Pyo3 is for us to write rust code that can then interact with our Python code? And it compiles this code into python code?

Reply:
the reason we use pyo3 indeed to interact between rust and python, specifically the goal for this library is to call rust code from our python, and we expose this rust code to our python in the form of a python package. pyo3 helps us bridge the gap between python and rust, it generates bindings for us so that we can call the rust code from the python, so there isn't any compilation into python, the rust code get compiled, there's some bindings so that python knows how to call specific functions that we expose, and when they get called, Rust code starts being executed. I'm not sure about the internals and how bindings work in general though